code/components/extra-natives-five/src/VehicleExtraNatives.cpp

```diff
@@ -866,6 +878,24 @@ static HookFunction initFunction([]()
 		};
 	};
 
+	fx::ScriptEngine::RegisterNativeHandler("GET_VEHICLE_GEAR_RATIO", [](fx::ScriptContext& context)
+	{
+		unsigned char gear = context.GetArgument<int>(1);
+		if (fwEntity* vehicle = getAndCheckVehicle(context, "GET_VEHICLE_GEAR_RATIO"))
+		{
+			context.SetResult<float>(*(float*)((char*)vehicle + VehicleGearRatioOffset + gear * sizeof(float)));
```

Comment:
You must ensure the user won't be able to read or write out of bounds. As of b3095, it seems the biggest possible amount of gears is `<= 10`. You need to check that the passed `gear` argument isn't bigger than 10. You can also use `*(uint8_t*)(CTransmission+6)` to get this value directly from memory.
```
// The same code block we're used to calculate "VehicleGearRatioOffset".
VehicleMaxGearOffset = CurrentGearOffset + 6;
```

Here's a pattern to the function where this max gear value is being set: `BA ? ? ? ? F3 0F 11 49`